### PR TITLE
[AMORO-2068] Upgrade the version description of Amoro Flink documenta…

### LINKED ISSUE #2068: [Subtask]: Upgrade the version description of Amoro Flink documentation
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

Amoro Flink Now Support Flink 1.17, so remove expired 1.12/1.14 support docs
And Flink QuickStart Demo Upgrade to Flink 1.17

### Parent issue

https://github.com/NetEase/amoro/issues/1998

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by czy006**
## Why are the changes needed?

- Upgrade the version description of Amoro Flink docs

Close #2068.

## Brief change log

- Upgrade the version description of Amoro Flink docs

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (docs)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -134,7 +134,7 @@ Amoro is built using Maven with Java 1.8 and Java 17(only for `trino` module).
 * To skip tests: `mvn -DskipTests package -P toolchain`
 * To package without trino module and JAVA 17 dependency: `mvn clean package -DskipTests -pl '!trino'`
 * To build with hadoop 2.x(the default is 3.x) `mvn clean package -DskipTests -Dhadoop=v2`
-* To indicate flink version for optimizer(the default is 1.14, 1.15 and 1.16 are available)
+* To indicate flink version for optimizer(the default is 1.15, 1.16 and 1.17 are available)
```
Comment (zhoujinsong) [inline README.md:137]: The Flink optimizer module supports specifying the dependent Flink version through profiles. Currently, versions 1.14/1.15/1.16 are supported, so there is no need to update the documentation. By the way, we can consider upgrading the version in the profile and then modifying the documentation here.

Comment (zhoujinsong) [reply README.md:137]: ref:https://github.com/NetEase/amoro/blob/master/ams/optimizer/flink-optimizer/pom.xml#L257

**Review by wangtaohz — COMMENTED** [review]
Thanks for your contribution, I left some comments.

File: README.md (hunk 1 above)
Comment (czy006) [reply README.md]: We can then keep in sync with the amoro flink connector version

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**